.github/workflows/ci.yml

```
@@ -0,0 +1,155 @@
+          -DCMAKE_CXX_FLAGS="-O3 -march=native -DNDEBUG"
+
+    - name: Build
+      run: cmake --build build
+
+    - name: Run Benchmarks
+      run: |
+        echo "=== Performance Test ==="
+        time cmake --build build --target test
+        echo "=== Memory Usage Test ==="
+        valgrind --tool=massif --massif-out-file=massif.out \
+          ./build/test_suite 2>/dev/null || true
```

YAMLlint (1.37.1)

[error] no new line character at the end of file

(new-line-at-end-of-file)